web/packages/teleport/src/JoinTokens/CreateJoinTokenDialog.tsx

```diff
@@ -0,0 +1,565 @@
+                  width="100%"
+                  textTransform="none"
+                  size="large"
+                  onClick={() => {
+                    reset(validator);
+                    onClose();
+                  }}
+                  disabled={false}
+                >
+                  Cancel
+                </ButtonSecondary>
+              </Flex>
+            </Box>
+          )}
+        </Validation>
+      </Box>
+    </Flex>
+  );
+};
+
+const RuleBox = styled(Box)`
+  border-color: ${props =>
+    props.theme.colors.interactive.tonal.neutral[0].background};
+  border-width: 2px;
+  border-style: solid;
+  border-radius: ${props => props.theme.radii[2]}px;
+
+  margin-bottom: ${props => props.theme.space[3]}px;
+
+  padding: ${props => props.theme.space[3]}px;
+`;
+
+const JoinTokenIAMForm = ({
+  tokenState,
+  onUpdateState,
+}: {
+  tokenState: NewJoinTokenState;
+  onUpdateState: (newToken: NewJoinTokenState) => void;
+}) => {
+  const rules = tokenState.iam;
+
+  function removeRule(index: number) {
+    const newRules = rules.filter((_, i) => index !== i);
+    const newState = {
+      ...tokenState,
+      iam: newRules,
+    };
+    onUpdateState(newState);
+  }
+
+  function setTokenRulesField(
+    ruleIndex: number,
+    fieldName: string,
+    value: string
+  ) {
+    const newState = {
+      ...tokenState,
+      [tokenState.method.value]: tokenState[tokenState.method.value].map(
+        (rule, i) => {
+          if (ruleIndex !== i) {
+            return rule;
+          }
+          return {
+            ...rule,
+            [fieldName]: value,
+          };
+        }
+      ),
+    };
+    onUpdateState(newState);
+  }
+
+  function addNewRule() {
+    const newState = {
+      ...tokenState,
+      iam: [...tokenState.iam, { aws_account: '' }],
+    };
+    onUpdateState(newState);
+  }
+
+  return (
+    <>
+      {rules.map((rule, index) => (
+        <RuleBox
+          key={index} // order does not change without updating the reference array
+        >
+          <Flex alignItems="center" justifyContent="space-between">
+            <Text fontWeight={700} mb={2}>
+              AWS Rule
+            </Text>
+
+            {rules.length > 1 && ( // at least one rule is required, so lets not allow the user to remove it
+              <ButtonIcon onClick={() => removeRule(index)}>
+                <Trash size={16} color="text.muted" />
+              </ButtonIcon>
+            )}
+          </Flex>
+          <FieldInput
+            label="Account ID"
+            rule={requiredField('Account ID is required')}
+            placeholder="AWS Account ID"
+            value={rule.aws_account}
+            onChange={e =>
+              setTokenRulesField(index, 'aws_account', e.target.value)
+            }
+          />
+          <FieldInput
+            label="ARN"
+            toolTipContent={`The joining nodes must match this ARN. Supports wildcards "*" and "?"`}
```

Comment:
does this need to be a specific role arn format? can we maybe provide example? like how do you use the question mark in the arn? 

eg (this is IAM role arn format)
- arn:aws:iam::account-id:role/role-name
- arn:aws:iam::account-id:role/*
- question mark?